### ISSUE TITLE
H2 enum values which contain brackets are not generated as Java literals

### ISSUE THREAD

**Issue by sco0ter** [jOOQ/jOOQ#12463]
### Expected behavior

All defined enum values are generated in the Java enum:

```java
public enum TestLanguage implements EnumType {
    German("German");
    Portuguese_Brazil("Portuguese (Brazil)");
    Russian("Russian");

    private final String literal;

    private TestLanguage(String literal) {
        this.literal = literal;
    }

    // ...
}
```

### Actual behavior

Only the enum value before the enum value which contains the bracket are generated:

```java
public enum TestLanguage implements EnumType {
    German("German");

    private final String literal;

    private TestLanguage(String literal) {
        this.literal = literal;
    }

    // ...
}
```

### Steps to reproduce the problem

database.sql:
```
CREATE TABLE `Test` (
                                id INTEGER PRIMARY KEY AUTO_INCREMENT,
                                language ENUM('German', 'Portuguese (Brazil)', 'Russian')
);
```

Maven configuration:
```xml
    <properties>
        <jooq.version>3.15.3</jooq.version>
    </properties>
...
        <plugin>
                <groupId>org.jooq</groupId>
                <artifactId>jooq-codegen-maven</artifactId>
                <version>${jooq.version}</version>
                <executions>
                    <execution>
                        <id>jooq-codegen</id>
                        <phase>generate-sources</phase>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <configuration>
                            <generator>
                                <database>
                                    <name>org.jooq.meta.extensions.liquibase.LiquibaseDatabase</name>
                                    <properties>
                                        <property>
                                            <key>scripts</key>
                                            <value>
                                                ${project.basedir}/src/main/resources/databaseChangelog.xml
                                            </value>
                                        </property>
                                    </properties>
                                </database>
                            </generator>
                        </configuration>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>org.jooq</groupId>
                        <artifactId>jooq-meta-extensions-liquibase</artifactId>
                        <version>${jooq.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
```

databaseChangelog.xml:
```
<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-2.0.xsd">

    <changeSet author="me" id="1">
        <sqlFile path="database.sql" relativeToChangelogFile="true"/>
    </changeSet>

</databaseChangeLog>
```

### Versions

- jOOQ: 3.15.3
- Java: 11
- Database (include vendor): Liquibase Generator
- OS: Windows
- JDBC Driver (include name if inofficial driver): ?

**Comment by lukaseder** [comment]
Thanks a lot for your report. I'll look into this shortly.

**Comment by lukaseder** [comment]
It doesn't fail using the `MySQLDatabase`

**Comment by lukaseder** [comment]
But I can reproduce it with the `DDLDatabase`, which uses the same underlying mechanism translating SQL to H2 behind the scenes, so that's a start.

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.5 (https://github.com/jOOQ/jOOQ/issues/12623), and 3.14.16 (https://github.com/jOOQ/jOOQ/issues/12624). Thanks again for your report!
